doc/develop/test/twister_statuses.rst

```diff
@@ -0,0 +1,206 @@
+
+Twister Status
+##############
+
+What is a Twister Status?
+=========================
+
+Twister Status formulates the current state of
+
+- ``Harness``
+- ``TestCase``
+- ``TestSuite``
+- ``TestInstance``
+
+in a comprehensive and easy-to understand way.
+In practice, most users will be interested in Statuses
+of Instances and Cases after the conclusion of their Twister runs.
+
+.. tip::
+
+   Nomenclature reminder:
+
+   .. tabs::
+
+      .. tab:: ``Harness``
+
+         ``Harness`` is a Python class inside Twister that allows us
+         to capture and analyse output from a program external to Twister.
+
+      .. tab:: ``TestCase``
+
+         ``TestCase`` is a piece of code that aims to verify some assertion.
+         It is the smallest subdivision of testing in Zephyr.
+
+      .. tab:: ``TestSuite``
+
+         ``TestSuite`` is a grouping of Test Cases.
+         One can modify Twister's behaviour on a per-Suite basis via ``testcase.yaml`` files.
+         Such grouped Cases should have enough in common
+         for it to make sense to treat them all the same by Twister.
+
+      .. tab:: ``TestInstance``
+
+         ``TestInstance`` is a Test Suite on some platform.
+         Twister reports typically report the results for Instances,
+         despite them being called "Suites" there.
+         As the distinction between them is not useful in this section,
+         whenever you read "Suite", assume the same for Instances.
+
+   More detailed explanation can be found :ref:`here <twister_tests_long_version>`.
+
+Possible Twister Statuses
+=========================
+
+.. list-table:: Twister Statuses
+   :widths: 10 10 70 10
+   :header-rows: 1
+
+   * - In-code
+     - In-text
+     - Description
+     - Where applicable
+   * - NOTRUN
+     - not run
+     - The test wasn't run because it was not runnable in current configuration.
+       It was, however, built correctly.
+     - Suite, Case
+   * - BLOCK
+     - blocked
+     - The test was not run because of an error or crash in the test suite.
+     - Case
+   * - FILTER
+     - filtered
+     - A static or runtime filter has eliminated the test from the list of tests to use.
+     - Suite, Case
+   * - ERROR
+     - error
+     - The test produced an error in running the test itself.
+     - Suite, Case
+   * - FAIL
+     - failed
+     - The test produced results different than expected.
+     - Suite, Case
+   * - PASS
+     - passed
+     - The test produced results as expected.
+     - Suite, Case
+   * - SKIP
+     - skipped
+     - Test was skipped by some other reason than previously delineated.
+     - Suite, Case
+
+**In-code** and **In-text** columns should the names of a given status.
+If one is not a Twister developer, the in-text name should suffice.
+It is the name used in the JSON reports.
+
+**Where applicable** column indicates where we might find such Statuses.
+Harness has been omitted for clarity, as it does not appear in final report files.
```

Comment:
these two sentences, esp. for Harness, might be moved to the 'Nomenclature reminder' module.

Reply:
Makes sense, especially with the table column change.